### PR TITLE
IDEMPIERE-5772 - Info Window Ctx - Paging Issue

### PR DESCRIPTION
**Pull request by PeterTakacs300**
Reopened JIRA ticket: https://idempiere.atlassian.net/browse/IDEMPIERE-5772

# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [ ] I have made corresponding changes to the documentation as follows:
  - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
  - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

**Comment by PeterTakacs300** [comment]
After the changes, the info window can put and preserve values in the ctx from other pages too (e.g. with the select all button or selecting a few lines manually, then switch to another page, and continue the selection there)

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/InfoPanel.java [hunk 1]
```diff
@@ -621,7 +621,7 @@ private void init()
 	/**
 	 * Contains the indexes of selected row, maintains the selection order
 	 */
-	protected ArrayList<Integer> m_rowSelectionOrder = new ArrayList<Integer>();
+	protected ArrayList<Object> m_rowSelectionOrder = new ArrayList<Object>();
```
[the remark this answers is not in the document]
Comment (PeterTakacs300) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/InfoPanel.java:624]: done

**Comment by PeterTakacs300** [comment]
Hi @hengsin, regarding your concern about this change: with Norbert we also discussed this topic, we chose the current approach based on that the processes can also take the selected rows from other pages as input - we wanted to preserve this UI logic for the context variables as well. Also, you can keep track of the number of selected rows with the message on the footer of the window.

**Comment by CarlosRuiz-globalqss** [comment]
Hi @PeterTakacs300, @norbertbede, @hengsin

Testing this I'm also a bit concerned about the user experience.

The getLastSelectedRow() is literally getting the "last" selected record as the last in the grid, not as the last selected by the user.

I think any user would suppose to get the information of the record they select, not from the record that is last in the grid, and even more confuse if the record is not visible as Heng Sin pointed.

So, for example if I open _Order Info_ and select the second row, it shows information about that order, and then I select the first row, it still shows the information about the second order.  I found that a bit confusing.

But I understand if solving that is problematic, because we would need to keep track of the really manually selected from the user for example in contrast to records selected using the "all" button or checbox.

I would like to suggest a different approach:
- fill the _IWInfo_ context variables **JUST** when one record is selected, if there are multiple records selected, then do not fill the _IWInfo_ variables, but of course still fill the _IWInfoIDs_Selected_
WDYT?

**Comment by hengsin** [comment]
This needs to be tested but I think getSelectedItem or getSelectedIndex should give you the current focus row.

**Comment by PeterTakacs300** [comment]
Hi @CarlosRuiz-globalqss, the getLastSelectedRow() method was intended to return the last selected by the user (this is what the m_rowSelectionOrder was meant for). It should work like that with the current merged version.

Looks like there is a bug now in the updateRowSelectionOrder() method after the latest changes, I need to review that

**Comment by PeterTakacs300** [comment]
Another issue found while testing: 
open an info window with no multi-selection ( = no process attached)
list some rows and select a row
set the parameters in a way so no rows will be displayed after re-query
press re-query

https://github.com/idempiere/idempiere/assets/93127072/6a52dc42-0139-44e8-8cc4-c6c26290196c

**Comment by PeterTakacs300** [comment]
Hi @hengsin, @CarlosRuiz-globalqss  - sorry for the inactivity. Both the above mentioned issues should be fixed now (showing ctx info of the last selected row by the user + the access nonexistent field error I reported lately).

The PR is ready for review.

**Comment by CarlosRuiz-globalqss** [comment]
Hi @hengsin - I tested this successfully.

**Review by hengsin — APPROVED** [review]
tested by Carlos

**Pull request merged**